ghost/stripe/lib/WebhookController.js

```diff
@@ -27,6 +27,25 @@ module.exports = class WebhookController {
             'invoice.payment_succeeded': this.invoiceEvent,
             'checkout.session.completed': this.checkoutSessionEvent
         };
+
+        // We pass the entire `deps` object to ensure all dependencies are available
+        // for services that may require dynamic access to multiple dependencies.
+        // This is particularly important in E2E tests where real services
+        // interact with a full set of dependencies (e.g., various repositories, services).
+        // Limiting injected dependencies may cause failures when certain dependencies
+        // are not explicitly passed but are expected by the services at runtime.
+        this.initializeServices(deps);
+    }
+
+    /**
+     * Initializes event services
+     * @param {object} deps 
+     */
+    initializeServices(deps) {
```

Comment:
Rather than initialise these services here, you can pass them in as dependencies - this gives you the full benefit of dependency injection!

In this case it would mean that the dependencies of the `StripeWebhookController` are just `webhookManager` and these three services! That would mean unit tests for the controller are a lot simpler, too

Reply:
Thanks for the suggestion. Made this change and it works great!